---
title: "Autism Project EDA"
author: "Carlos Ramírez"
date: "August 4, 2017"
output: html_document
---


## Exploratory Data Analysis on Autism SNP data

Illumina raw results are provided as input files for Plink, Partek and GenomeStudio. Additionally, three important files can be observed named: i)
"*_FinalReport.txt", ii) "\*.opa", and iii) "SampleSheet.\*.csv". 
FinalReport.txt file contains info about the nucleotide composition for each SNP in every sample. The ".opa" file summarizes SNP genomic information such as chromosome location and position. Finally, "sampleSheet.csv" seems to contain patient sample ID.  
This report shows data processing regarding these three files.

```{r}
## loading data from local repositories
## local directory
pathReport <- "~/dataBases/GoldenGate_U15_025 AT/FinalReport/U15_025_ID_FinalReport.txt"
pathOpa <- "~/dataBases/GoldenGate_U15_025 AT/GenomeStudio/GS0014416-OPA.opa"
pathSampleSheet <- "~/dataBases/GoldenGate_U15_025 AT/GenomeStudio/SampleSheet_ID.csv"

## loading into R
freport <- read.csv(file = pathReport,
            sep = "\t",
            header = TRUE,
            skip = 9,
            stringsAsFactors = FALSE,
            colClasses = "character")

opa <- read.csv(file = pathOpa,
                skip = 12,
                header = TRUE)

sample <- read.csv(file = pathSampleSheet,
         skip = 8,
         header = TRUE)

## display files composition
lapply(list(freport[1:10],opa,sample), names)
```

As can be inferred from inspecting the files and from the given output FinalReport.txt file contains in its first field (column) called "X" the illumina SNP ID. The rest of the fields (AU_118_1, AU_103_1, ...) corresponds to sample IDs also present in the sampleSheet.csv file. The entries of the matrix given in this file contains observed nucleotides. 

In the rest of the report frequency of nucleotide markers are obtained from
finalReport.txt file. Using the information on SNP location from .opa file a bar graph of frequencies for chromosome 22 is constructed.

The next code shows data filtering process.

```{r}
## keeping only SNP data
for (i in 2:ncol(freport)) {
        freport[,i] <- gsub("\\|.*","",freport[i,2:ncol(freport)])
}

## recording existing SNPs
bag <- c()
for (j in 2:ncol(freport)) {
        for (i in 1:nrow(freport)) {
                if ( ! ( freport[i,j] %in% bag ) ) {
                        bag <- c(bag,freport[i,j])
                }
        }
}
bag <- sort(bag)

## matrix of frequencies definition
frequencies <- matrix(data = 0,
                      nrow = nrow(freport),
                      ncol = length(bag))
colnames(frequencies) <- bag


## getting frequencies from haplotypes
for (i in 1:nrow(freport)) {
        res <- table(gsub("\\|.*","",freport[i,2:ncol(freport)]))
        frequencies[i,names(res)] <- res
}

frequencies.df <- as.data.frame(frequencies)
frequencies.df$"X" <- freport[,"X"]
head(frequencies.df)
```

The next code relates so created frequencies.df dataframe to their chromosome
location located in the .opa file.

```{r}
chrPosition <- opa[,c("Name","CHR")]
names(chrPosition) <- c("X","CHR")
frequencies.df <- merge(frequencies.df,chrPosition,by="X",all = FALSE)
head(frequencies.df)
```

Finally, as an example the next plot shows the frequencies of each marker located in the chromosome 20 in the sample population.

```{r}
library(ggplot2)
library(reshape2)

chrNumber <- 20
n <- as.character(chrNumber)
frequencies.m <- melt(frequencies.df[frequencies.df$CHR==n,])
g <- ggplot(data = frequencies.m, aes(x = X,y = value,fill=variable))
g <- g + scale_color_manual(values=variable)
g <- g + geom_bar(stat="identity",width = 1)
g <- g + theme(axis.line=element_blank(),
               axis.text.x=element_blank(),
               axis.title.x=element_blank(),
               panel.background=element_blank(),
               panel.border=element_blank(),
               panel.grid.major=element_blank(),
               panel.grid.minor=element_blank(),
               plot.background=element_blank())
plot(g)
```

## Conclusiones

El análisis exploratorio preliminar de los datos muestra que la información
puede ser obtenida a partir de varios archivos proveídos por Illumina como los archivos de entrada de Plink, Partek y genomeStudio. Adicionalmente, también son proveídos resúmenes con los marcadores observados para cada SNP en cada muestra en el archivo FinalReport.txt en cual se basó este reporte para obtener las frequencias de marcadores en la población muestreada. 

Como perspectiva inmediata se tiene un análisis exploratorio de los datos
partiendo de los archivos de entrada a Plink, Partek y GenomeStudio. Después de esto se procederá al análisis descriptivo de los datos mediante por ejemplo la observación de marcadores genómicos sobrerrepresentados en los pacientes cuando se tengan los datos sobre los controles. Otra perspectiva es analizar las bases de datos disponibles publicamente.



